examples/gogo/command/src/main/java/org/osgi/enroute/examples/gogo/command/upstream/annotations/RequireCommand.java

```diff
@@ -0,0 +1,30 @@
+// Needs to be moved upstream
+
+package org.osgi.enroute.examples.gogo.command.upstream.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.osgi.annotation.bundle.Requirement;
+import org.osgi.annotation.bundle.Requirements;
+
+/**
+ * This annotation can be used to require the OSGi command implementation.
+ * It can be used directly, or as a meta-annotation.
+ * <p>
+ * This annotation is applied to the command property annotations meaning that it does not normally need to be
+ * applied to Declarative Services components itself.
+ */
+@Documented
+@Retention(RetentionPolicy.CLASS)
+@Target({ ElementType.TYPE, ElementType.PACKAGE })
+@Requirements(value = { //
+        @Requirement(namespace = "osgi.wiring.bundle", name = "org.apache.felix.gogo.shell"), //
+        @Requirement(namespace = "osgi.identity", name = "org.apache.felix.gogo.command"), //
+})
```

Comment:
@timothyjward I gave it a try, can you comment please?
I would like to use `osgi.implementation` but as long as Gogo doesn't provide some capability, we have to require the specific bundle -- this should be changed by Gogo upstream first. It seems there is already a Jira for Felix Gogo.
If my reading has been correct `osgi.wiring.bundle` and `osgi.identity` can be used to require a specific bundle. Which one should be preferred?

Reply:
`osgi.wiring.bundle` is the equivalent  of `Require-Bundle` while `osgi.identity` is a requirement for a bundle with an identity. I would suggest using `osgi.identity` for now because we don't want want to give the impression  that we need a require-bundle semantic.